src/starknet.rs

```diff
@@ -584,10 +542,14 @@ pub(crate) mod handler {
         nonce: Felt252Abi,
     }
 
+    /// A C ABI Wrapper around the StarknetSyscallHandler
+    ///
+    /// It contains pointers to functions which can be called through MLIR based on the field offset
+    /// The functions convert C ABI structures to the Rust equivalent and calls the wrapped implementation.
     #[repr(C)]
     #[derive(Debug)]
     pub struct StarknetSyscallHandlerCallbacks<'a, T> {
-        self_ptr: &'a mut T,
+        pub self_ptr: &'a mut T,
```

Comment:
is there a reason for making this public?

Reply:
Originally I was accessing `self_ptr` through `StarknetSyscallHandlerCallbacks` instead of dereferencing the pointer. Now there is no need to have it public, i'm going to correct it.

Reply:
Done